packages/react/src/hooks/useSizeObserver.js

```diff
@@ -0,0 +1,23 @@
+import useResizeObserver from 'use-resize-observer';
+import warning from 'warning';
+
+import { browserSupports } from '../utils/componentUtilityFunctions';
+
+const useSizeObserver = (ref) => {
+  if (!browserSupports('ResizeObserver')) {
```

Comment:
is the only purpose of this function to add the warning?

Reply:
That's the correct. And, that's why I suppressed the "rules-of-hooks" eslint message. There's no concept of a ErrorBoundary within a hook, so I thought it was pretty to at least print the warning rather than throwing an error.